### PR TITLE
save work

### PR DESCRIPTION
**Pull request by Allensy**

### CODE REVIEW

File: src/components/diagnosticExercise/controllers/workoutsDiagnosticExercise.controller.js [hunk 1]
```diff
@@ -7,10 +7,12 @@
...
                   $log, $state, ExerciseResultSrv, ExerciseTypeEnum, $q, $timeout, ZnkExerciseUtilitySrv,
                   $rootScope, ExamTypeEnum, exerciseEventsConst, $filter, SubjectEnum, znkAnalyticsSrv, StatsEventsHandlerSrv,
-                  $translate, ExerciseReviewStatusEnum) {
+                  $translate, ExerciseReviewStatusEnum, CategoryService) {
             'ngInject';
             var self = this;
-            this.subjectId = exerciseData.questionsData.subjectId;
+            var questionCategoryForSubjectId = exerciseData.questionsData.categoryId || exerciseData.questionsData.categoryId2;
+
+            this.subjectId = CategoryService.getCategoryLevel1ParentByIdSync(questionCategoryForSubjectId);
```
Comment (Ariel-W) [inline src/components/diagnosticExercise/controllers/workoutsDiagnosticExercise.controller.js]: Use the CategorySrv get subject fn (w. array of categories)

File: src/components/infraWebAppZnkExercise/directives/answerExplanation/answerExplanation.directive.js [hunk 2]
```diff
@@ -25,6 +25,8 @@
                     var ngModelCtrl = ctrls[1];
                     var viewMode = questionBuilderCtrl.getViewMode();
                     var question = questionBuilderCtrl.question;
+                    var questionCategoryForSubjectId = question.categoryId || question.categoryId2;
+                    var questionSubjectId = CategoryService.getCategoryLevel1ParentByIdSync(questionCategoryForSubjectId);
```
Comment (Ariel-W) [inline src/components/infraWebAppZnkExercise/directives/answerExplanation/answerExplanation.directive.js]: Use the CategorySrv get subject fn (w. array of categories)

File: src/components/workoutsRoadmap/controllers/workoutsRoadmap.controller.js [hunk 3]
```diff
@@ -36,7 +36,9 @@
 
             function _isFirstWorkoutStarted() {
                 var firstWorkout = vm.workoutsProgress[0];
```
Comment (Ariel-W) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmap.controller.js]: Revert.  Simply use subjectId

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapBasePreSummary.controller.js [hunk 4]
```diff
@@ -17,10 +17,12 @@
                 data.roadmapCtrlActions.freezeWorkoutProgressComponent(true);
 
                 var currentWorkout = data.exercise;
+                var currentWorkoutCategoryId = currentWorkout.categoryId || currentWorkout.categoryId2;
+                var currentWorkoutSubjectId = CategoryService.getCategoryLevel1ParentByIdSync(currentWorkoutCategoryId);
 
                 var subjectToIgnoreForNextDaily;
                 if (currentWorkout.workoutOrder !== DIAGNOSTIC_ORDER) {
-                    subjectToIgnoreForNextDaily = currentWorkout.subjectId;
+                    subjectToIgnoreForNextDaily = currentWorkoutSubjectId;
```
Comment (Ariel-W) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapBasePreSummary.controller.js]: Revert. Simply use subjectId

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapDiagnostic.controller.js [hunk 5]
```diff
@@ -2,13 +2,14 @@
...
-        function ($state, ExerciseStatusEnum, data, $timeout) {
+        function ($state, ExerciseStatusEnum, data, $timeout, CategoryService) {
             'ngInject';
             //  fixing page not rendered in the first app entrance issue
             $timeout(function () {
                 switch (data.diagnostic.status) {
                     case ExerciseStatusEnum.COMPLETED.enum:
-                        var isFirstWorkoutStarted = angular.isDefined(data.workoutsProgress[0].subjectId);
+                        var workoutSubjectId = CategoryService.getCategoryLevel1ParentSync(data.workoutsProgress[0]);
+                        var isFirstWorkoutStarted = angular.isDefined(workoutSubjectId);
```
Comment (Ariel-W) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapDiagnostic.controller.js]: Revert. Simply use subjectId

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js [hunk 6]
```diff
@@ -50,7 +50,8 @@
                 if (prevWorkout.status === ExerciseStatusEnum.COMPLETED.enum) {
                     if (!currWorkout.personalizedTimes) {
                         if (currWorkout.workoutOrder !== FIRST_WORKOUT_ORDER) {
-                            subjectsToIgnore = prevWorkout.subjectId;
+                            var prevWorkoutSubjectId = CategoryService.getCategoryLevel1ParentSync(prevWorkout);
+                            subjectsToIgnore = prevWorkoutSubjectId;
```
Comment (Ariel-W) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js]: Revert. Simply use subjectId

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js [hunk 7]
```diff
@@ -76,7 +77,8 @@
                 var subjectNum = SubjectEnum.getEnumArr().length;
 
                 return function () {
-                    usedSubjects.push(currWorkout.subjectId);
+                    var currentWorkoutSubjectId = CategoryService.getCategoryLevel1ParentSync(currWorkout);
+                    usedSubjects.push(currentWorkoutSubjectId);
```
Comment (Ariel-W) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js]: Revert. Simply use subjectId

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js [hunk 8]
```diff
@@ -152,7 +156,8 @@
 
                 if (vm.workoutsByTime) {
                     vm.selectedWorkout = vm.workoutsByTime[newSelectedTime];
-                    currWorkout.subjectId = vm.selectedWorkout.subjectId;
+                    var selectedWorkoutSubjectId = CategoryService.getCategoryLevel1ParentSync(vm.selectedWorkout);
```
Comment (Ariel-W) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js]: Revert. Simply use subjectId

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js [hunk 9]
```diff
@@ -99,6 +101,8 @@
 
             vm.startExercise = function(){
                 var selectedWorkout = angular.copy(vm.selectedWorkout);
+                var selectedWorkoutSubjectId = CategoryService.getCategoryLevel1ParentSync(selectedWorkout);
+                selectedWorkout.subjectId = selectedWorkoutSubjectId;
```
Comment (Ariel-W) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js]: Revert. Simply use subjectId

File: src/components/znkSummary/directives/znkSummaryTimeline.directive.js [hunk 10]
```diff
@@ -7,13 +7,14 @@
...
-        controller: function(SubjectEnum) {
+        controller: function(SubjectEnum, CategoryService) {
             'ngInject';
 
             var vm = this;
 
             vm.seenSummary = vm.exerciseData.exerciseResult.seenSummary;
-            vm.currentSubjectId = vm.exerciseData.exercise.subjectId;
+            
+            vm.currentSubjectId = CategoryService.getCategoryLevel1ParentSync(vm.exerciseData.exercise);
```
Comment (Ariel-W) [inline src/components/znkSummary/directives/znkSummaryTimeline.directive.js]: revert and use vm.exerciseData.exerciseResult.subjectId

File: src/components/completeExercise/directives/completeExerciseIntroSection/completeExerciseIntroSection.directive.js [hunk 11]
```diff
@@ -8,19 +8,22 @@
...
                 completeExerciseIntroCtrl: '^completeExerciseIntro'
             },
-            controller: function ($filter) {
+            controller: function ($filter, CategoryService) {
                 'ngInject';
 
                 this.$onInit = function(){
                     var exerciseParentContent = this.completeExerciseIntroCtrl.getExerciseParentContent();
                     var exerciseContent = this.completeExerciseIntroCtrl.getExerciseContent();
+                    var categoryIdForSubjectId = exerciseContent.catgoryId || exerciseContent.catgoryId2;
```
Comment (Allensy) [inline src/components/completeExercise/directives/completeExerciseIntroSection/completeExerciseIntroSection.directive.js]: fixed

File: src/components/diagnosticExercise/controllers/workoutsDiagnosticExercise.controller.js (hunk 1 above)
Comment (Allensy) [inline src/components/diagnosticExercise/controllers/workoutsDiagnosticExercise.controller.js]: fixed

File: src/components/infraWebAppZnkExercise/directives/answerExplanation/answerExplanation.directive.js (hunk 2 above)
Comment (Allensy) [inline src/components/infraWebAppZnkExercise/directives/answerExplanation/answerExplanation.directive.js]: fixed

File: src/components/workoutsRoadmap/controllers/workoutsRoadmap.controller.js (hunk 3 above)
Comment (Allensy) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmap.controller.js]: fixed

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapBasePreSummary.controller.js (hunk 4 above)
Comment (Allensy) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapBasePreSummary.controller.js]: fixed

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapDiagnostic.controller.js (hunk 5 above)
Comment (Allensy) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapDiagnostic.controller.js]: fixed

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js (hunk 6 above)
Comment (Allensy) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js]: fixed

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js (hunk 7 above)
Comment (Allensy) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js]: fixed

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js (hunk 9 above)
Comment (Allensy) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js]: fixed

File: src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js (hunk 8 above)
Comment (Allensy) [inline src/components/workoutsRoadmap/controllers/workoutsRoadmapWorkoutIntro.controller.js]: fixed

File: src/components/znkSummary/directives/znkSummaryTimeline.directive.js (hunk 10 above)
Comment (Allensy) [inline src/components/znkSummary/directives/znkSummaryTimeline.directive.js]: fixed

**Pull request merged**